src/tools/perf/perftest.c

```diff
@@ -460,8 +460,8 @@ static ucs_status_t parse_message_sizes_params(const char *optarg,
     }
     ++token_num;
 
-    free(params->msg_size_list); /* free previously allocated buffer */
-    params->msg_size_list = malloc(sizeof(*params->msg_size_list) * token_num);
+    params->msg_size_list = realloc(params->msg_size_list,
```

Comment:
i'm not sure if we should care about this but need to use other pointer to check realloc return value otherwise there is a memory leak in case of realloc failure.